src/main/java/seedu/mtracker/commands/DeleteCommand.java

```diff
@@ -0,0 +1,32 @@
+package seedu.mtracker.commands;
+
+import seedu.mtracker.asserthelpers.AssertCommandHelpers;
+import seedu.mtracker.model.Instrument;
+import seedu.mtracker.ui.TextUi;
+
+public class DeleteCommand extends Command {
+    public static final String COMMAND_WORD = "delete";
+    public static final int UNINITIALISED_INDEX = -1;
+    private int index;
+
+    public DeleteCommand() {
+        index = UNINITIALISED_INDEX;
+    }
+
+    public void setIndex(int index) {
+        this.index = index;
+    }
+
+    public int getIndex() {
+        return index;
+    }
+
+    @Override
+    public String execute() {
+        AssertCommandHelpers.assertIndexWithinBounds(instrumentManager.getInstruments().size(), index);

```

Comment:
can implement a getSize() method inside instrumentManager instead? Might be useful for the other functions too